src/transformations/RenameEntitiesTransformation.cpp

```diff
@@ -96,8 +99,12 @@ string RenameEntitiesVisitor::randomSnakeCaseName() {
     return newName;
 }
 
+string RenameEntitiesVisitor::hashName(string * name) {
+    return hash_prefix + "_" + std::to_string(std::hash<std::string>{}(*name));
+}
+
 void RenameEntitiesVisitor::processVarDecl(Decl * decl, string * name) {
-    string randomName = test ? "test_" + *name : randomSnakeCaseName();
+    string randomName = test ? "test_" + *name : with_hash ? hashName(name) : randomSnakeCaseName();
```

Comment:
I guess multiple single-line `if` statement is quite not easy to read. I suggest writing in the following manner:
```
string randomName;
if (test) {
   randomName = "test_" + *name;
} else {
   randomName = with_hash ? hashName(name) : randomSnakeCaseName();
}
```